### DISCUSSION TITLE
[SHARE] Custom Maps made with RoadRunner, Imported and Tested with CARLA_0.9.14

### DISCUSSION THREAD

**Discussion by csonthomisi** [carla-simulator/carla#7181] [Show and tell]
Dear Reader!

Feel free to use these Custom generated Maps if you need one!

[GIT CARLA_Custom_Maps](https://github.com/csonthomisi/CARLA_Custom_Maps.git)

Recently I uploaded 2 Maps:
![Example](https://github.com/carla-simulator/carla/assets/48515100/50d3701c-553b-470b-be65-6332ae718804)
![Example_2](https://github.com/carla-simulator/carla/assets/48515100/fae97158-85e5-4bd1-b6cc-c1783750d2f1)

**Comment by NnaYelsel** [comment]
Hello @csonthomisi ,

Do these roads have a slope ? or is everything at altitude 0 ? 

Best regards

**Reply by csonthomisi** [reply]
These roads are totally flat, but if you need something specific I may be able to create for you!

**Reply by NnaYelsel** [reply]
Really nice from you part to propose it.

I posted a discussion here : https://github.com/carla-simulator/carla/discussions/7284 if you want more context.

I want to create a map with some hills / slopes and tried several ways to do it but without success. My final goal is to have a whole nice map but for the moment, event a basic map with a slope could already be good. Did you use RoadRunner to do these ones ? 

If you could try something basic to see if that is possible ? And also if you have some .fbx to share, that could help me a lot. 

Best regards.

**Reply by csonthomisi** [reply]
Sorry for the late reply..
I made a map with elevation, I hope you thought something like this:
[CARLA_Custom_Maps](https://github.com/csonthomisi/CARLA_Custom_Maps)
The name of the map is "Elevation_3"

**Reply by NnaYelsel** [reply]
That's really nice, thank you, it mean that it's possible to get hills :) 
I struggle to load it on Carla because I'm on windows...

Can you send me the files .xodr and .fbx that you obtained just after Roadrunner ? I'm trying to understand the map creation pipeline. Did you parametrize the hills on roadrunner ? 

Thank you again to provide some support about map creation.

**Reply by csonthomisi** [reply]
I made it with ubuntu and 0.9.14 version, and I tested to import it to a Packed Version (0.9.14) of Carla.
I also uploaded to the repository the RoadRunner exported files (.xodr & .fbx) [REPO](https://github.com/csonthomisi/CARLA_Custom_Maps)
I only used the RoadRunner for hill creaton and also for any textures.

**Reply by NnaYelsel** [reply]
Thank you, !  I'm trying to generate the packages for Windows but still some problems. 

I will send it here it I can provided them.

**Reply by NnaYelsel** [reply]
I could create the map Example_2 as archive for Windows, are you interested that I add this on your repo ?

**Reply by csonthomisi** [reply]
I would really appreciate it if you would!

**Reply by NnaYelsel** [reply]
Can you open a push access for me ?
![image](https://github.com/carla-simulator/carla/assets/49529118/c18e37f8-0819-4e5f-841d-c508c775e67a)

**Reply by bitvijays** [reply]
@csonthomisi Just curious, did you created these maps in RoadRunner? Did you had any success with Truevision? Mainly I am looking for a 3-lane circular road representing a motorway and being able to load that in Carla.

@NnaYelsel Did you had any lucky with uploading the @csonthomisi maps into the Carla (windows version)?

**Reply by csonthomisi** [reply]
@bitvijays Yeah, I only used the RoadRunner yet. I did not try the Truevision.
If you would like to ask for a new map, can you create an Issue at [https://github.com/csonthomisi/CARLA_Custom_Maps](https://github.com/csonthomisi/CARLA_Custom_Maps) ?

Thank you!

**Reply by NnaYelsel** [reply]
@bitvijays yes I did !

@csonthomisi , I have a trial version of Roadrunner and I can hardy create elevation on a surface (I use surface tool), It's really sharp. How did you do it on your side ?

**Comment by csonthomisi** [comment]
I made and uploaded a third map!
![Elevation_3](https://github.com/carla-simulator/carla/assets/48515100/8762da37-7be3-41a6-8e7f-0df1063d369f)
It has some slopes.

**Comment by abdellatifzaarour** [comment]
hello i wish that you are all doing weeel pls i need your help
 i need a long straight highway for my autonomous driving test to use it with carla.

**Comment by arscode** [comment]
Hi @csonthomisi, I am using CARLA 0.9.15 and 0.9.16 under (prebuilt) Ubuntu 20.04. Do your maps work with this CARLA version? I have a custom-made map, but whenever I am following the manual to ingest the map using the Docker method, it gives me several errors. and mostly on during ` make import ARGS="--packages=Map_custom"`. I have tried different ways and written dockerfile on my own, but somehow it's not getting through. I just want to make that tar.gz so that I can import it. Which method did you take to prepare your map assets?

**Reply by csonthomisi** [reply]
I haven’t tested these maps with CARLA 0.9.15 or 0.9.16, so I can’t guarantee full compatibility with those versions.

What I can confirm is that the maps were prepared for the packaged (prebuilt) version of CARLA, not the editor version.

For the packaged version, I did not use the Docker-based import pipeline. Instead, I prepared the assets into a .tar.gz file, placed it into the Import folder, and then used the ImportAssets.sh script to load the map.

Regarding the Docker method (`make import ARGS="--packages=Map_custom"`), I didn’t use that workflow myself, so I’m not sure what might be causing your errors there.

If you are working with the editor version, one possible workaround is to manually extract the assets into the appropriate folders under Unreal/Content, but I haven’t verified this approach for newer CARLA versions.

If you can share the specific error messages from the `make import` step, I might be able to help more.

**Reply by arscode** [reply]
I have the fbx and xodr file and other asset files. Without using Docker, how did you create that tar.gz? I am also using pre-built version of CARLA.

**Reply by csonthomisi** [reply]
For map package generation I built the Unreal Editor with CARLA project not in docker.

**Reply by arscode** [reply]
> For map package generation I built the Unreal Editor with CARLA project not in docker.

Ok thanks. you have used the procedure for CARLA built from source, I guess
https://carla.readthedocs.io/en/0.9.15/tuto_M_add_map_source/
